Responsive Schedule Page with Current and Next

Danzibob opened the pull request:
Functionally it's there, but could still do with some styling (not sure what though)
Might make sense to hide current and next on mobile
Desktop:
![image](https://user-images.githubusercontent.com/19689781/38209113-30370e1e-36ab-11e8-8e56-c70eb3b4fc17.png)
![image](https://user-images.githubusercontent.com/19689781/38209121-3eb33a1c-36ab-11e8-84cb-38852893c6d7.png)
Mobile:
![image](https://user-images.githubusercontent.com/19689781/38209149-55106bd6-36ab-11e8-974e-60591e4f66d3.png)
![image](https://user-images.githubusercontent.com/19689781/38209173-6613ec0a-36ab-11e8-8605-058beed865bb.png)

itstombellis committed:
Initial layout idea

itstombellis committed:
Attempt at adding colour to schedule

itstombellis committed:
Implementes structure

itstombellis committed:
Use Mixins and Rename Gold to Flagship

Brookke committed:
Basics schedule redesign

Danzibob committed:
Flipped table and created basic layout

Mysterious footer is somehow in the middle of the table...

Danzibob committed:
Actually responsive stuff, woo

Danzibob committed:
re-added time labels for rows; other polishing

Danzibob committed:
Add current and next to schedule

mstratford commented:
Certainly looks better :) Will look closely later :)

Danzibob committed:
Appease Codacy a wee bit

itstombellis commented:
It looks really nice. 

I am not a fan of the blank space where jukebox is on air, it implies that nothing is on.

Brookke commented:
Looks good :) Would it not be better to highlight the current show that is on in the schedule than have the current and next box there? 

I also wonder if we could make it so that we only display one day on mobile and then allowing scrolling/buttons that change the day to the next

Brookke commented on sass/elements/_schedule.scss:
```diff
@@ -69,3 +84,33 @@ tr {
   @include schedule-block(tint($speech-green,60%),$white);
 }
+
+@media (max-width: 767px) {
```
change to @include media-breakpoint-up(md) for consistency

mstratford commented:
At least saying "URY jukebox" within the blue space would be nice, without the white box.

Danzibob committed:
Un-hid sustainer boxes, hid CaN on mobile

Danzibob commented:
Brought back sustainer boxes:
![image](https://user-images.githubusercontent.com/19689781/38255251-3bb5bcd8-3753-11e8-8fda-fd087456d20a.png)
CaN hidden on mobile:
![image](https://user-images.githubusercontent.com/19689781/38255296-60bd7228-3753-11e8-9bb1-04fe19f3cf71.png)

@Brookke I intend to add something to highlight the current show on the schedule, but I want that to be in a seperate PR with the JS for updating Current and Next, since that seems more fitting. Also means that if I don't get that done on time, this is still pull-able. Would also like to add a button to jump to the current day on mobile, but that will also come with the JS. 

Personally (as a user) I'd prefer to have all the days on one scroll-able page, but I'm open to changing it depending on what others think.

mstratford commented:
@Danzibob If the JS is added to scroll to current day on start on mobile, and maybe some semi animated arrows, i'm happy. If you want to do that in another PR, sure!

mstratford commented:
Something I'd like to see fixed here is maybe putting the "for date - date" on same line as "Schedule" title. Also, bonus points for making the empty week look nicer :)

Brookke commented:
@mstratford I kinda like it underneath as a subtitle ahah, but I would drop the “for”

Brookke commented:
@Danzibob you can do the jumping without any js, see http://www.echoecho.com/htmllinks08.htm you can also use id's rather than names, might be more preferable here.

Brookke commented:
![5a0e4451-0cef-4df0-89f1-3b604338abc3](https://user-images.githubusercontent.com/10200358/38277048-f95fdd00-378e-11e8-8ad9-4770f6ebf377.jpeg)
Added for reference. I also agree with having the whole box clickable

Brookke commented:
@Danzibob re scrolling vs continuous, okay leave it as is in this pr, something to consider at a later date

Brookke commented:
Also out of scope for this pr, but would it be easier to swap to using lists long term? May be easier to style etc

itstombellis commented:
I wonder if dropping the background colour from the schedule might help to make it look cleaner? 

<img width="1162" alt="screen shot 2018-04-04 at 10 41 18" src="https://user-images.githubusercontent.com/16823823/38300518-e9608d58-37f4-11e8-8ac2-077510916e47.png">

Danzibob committed:
remove bluc schedule background on desktop

Danzibob committed:
Change header colors to match

Danzibob committed:
make whole box clickable

Danzibob committed:
some config-related atrocities

Danzibob committed:
Merge remote-tracking branch 'URY/development' into danzi-schedule

Danzibob committed:
Made schedule deal with straddled days properly
Straddled weeks still don't work though...

Danzibob committed:
Handle week straddling better
Still doesn't show the monday morning segment but that will probs need a decent rewrite

Danzibob committed:
Add Tom's colorizer code; Appease codacy

Danzibob committed:
remove print statement

Brookke commented:
@Tombellis yeah i that was the original design, i think i prefer it too. It would look even better with the grid too but that is nearly impossible to do without designing the schedule in a canvas :(

Brookke commented on views/schedule_week.tmpl:
```diff
@@ -63,12 +63,9 @@
               {{if ne .RowSpan 0}}
-              <td rowspan="{{.RowSpan}}" class="schedule-timeslot schedule-blocks-{{.Item.Block}}" title="{{stripHTML .Item.Desc}}">
-                {{if .Item.PageURL}}
-                  <a href="{{url .Item.PageURL}}">{{.Item.Name}}</a>
```
In removing the link href, if JS isnt loaded on the page, there will be no way to get to shows

Brookke commented on sass/elements/_schedule.scss:
```diff
@@ -0,0 +1,163 @@
+  @include schedule-block(tint($speech-green, 60%), $white);
+}
+
+@media screen and (max-width: 767px) {
```
Consider using the breakpoint constants here, see https://github.com/UniversityRadioYork/2016-site/blob/development/sass/base/_utils.scss#L15 for an example

Brookke commented on views/schedule_week.tmpl:
```diff
@@ -28,42 +33,52 @@
-          {{if ne .RowSpan 0}}
-          <td rowspan="{{.RowSpan}}" class="schedule-timeslot schedule-blocks-{{.Item.Block}}" title="{{stripHTML .Item.Desc}}">
-            {{if .Item.PageURL}}
-              <a href="{{url .Item.PageURL}}">{{.Item.Name}}</a>
```
In removing the link completely, if JS isnt enabled there would be no way to click through to shows

Danzibob committed:
fix weird white line at bottom of schedule page

Danzibob committed:
use media breakpoints and re-add JS-free links

Danzibob committed:
re-fix straddlesDay function

Danzibob committed:
remove debugs

Danzibob commented:
Feels pretty finished other than the ellipses right now. There _is_ still a bug where week-straddling shows dont appear on monday morning, but this is a very rare case and would require a significant rewrite. Furthermore, the start and end of term should be factored in to display "off-air" rather than jukebox. Again this needs a major rewrite. When this PR gets merged I will create issues for these... issues, if it's decided to leave them.

Danzibob committed:
Final Appeasement
...hopefully

Danzibob committed:
Fix Oversized CaN

itstombellis commented:
The now and next looks much better that size.
